agent_build_refactored/prepare_agent_filesystem.py

```diff
@@ -0,0 +1,393 @@
+
+
+def build_agent_base_files(
+    output_path: pl.Path,
+    install_type: str,
+    version: str = None,
+    frozen_binary_path: pl.Path = None,
+    copy_agent_source: bool = False,
+):
+    """
+    Create directory with agent's core files.
+    :param output_path: Output path for the root of the agent's base files.
+    :param install_type: String with install type of the future package.
+        See 'install_info' in scalyr_agent/__scalyr__.py module.
+    :param version: Version string to assign to the future package, uses version from the VERSION file if None.
+    :param frozen_binary_path: Path to frozen binaries, if specified, then those binaries will be used in the
+        'bin' folder. Excludes 'copy_agent_source'.
+    :param copy_agent_source: If True, then agent's source code is also copied into the '<output_path>/py' directory.
+        In opposite, it is expected that a frozen binaries will be placed instead of source code later.
+    """
+    if output_path.exists():
+        shutil.rmtree(output_path)
+
+    output_path.mkdir(parents=True)
+
+    # Copy the monitors directory.
+    monitors_path = output_path / "monitors"
+    shutil.copytree(SOURCE_ROOT / "monitors", monitors_path)
+    recursively_delete_files_by_name(output_path / monitors_path, "README.md")
+
+    # Add VERSION file.
+    result_version_path = output_path / "VERSION"
+    if version:
+        result_version_path.write_text(version)
+    else:
+        shutil.copy2(SOURCE_ROOT / "VERSION", output_path / "VERSION")
+
+    # Create bin directory with executables.
+    bin_path = output_path / "bin"
+
+    if frozen_binary_path:
+        shutil.copytree(frozen_binary_path / ".", bin_path)
+
+    elif copy_agent_source:
+        bin_path.mkdir()
+        source_code_path = output_path / "py"
+
+        shutil.copytree(SOURCE_ROOT / "scalyr_agent", source_code_path / "scalyr_agent")
+
+        agent_main_executable_path = bin_path / "scalyr-agent-2"
+        agent_main_executable_path.symlink_to(
+            pl.Path("..", "py", "scalyr_agent", "agent_main.py")
+        )
+
+        # Copy wrapper script for removed 'scalyr-agent-2-config' executable for backward compatibility.
+        shutil.copy2(
+            AGENT_BUILD_PATH / "linux/scalyr-agent-2-config",
+            bin_path,
+        )
+
+        # Write install_info file inside the "scalyr_agent" package.
+        install_info_path = source_code_path / "scalyr_agent" / "install_info.json"
+
+        install_info = get_install_info(install_type=install_type)
+        install_info_path.write_text(json.dumps(install_info))
+
+        # Don't include the tests directories.  Also, don't include the .idea directory created by IDE.
+        recursively_delete_dirs_by_name(
+            source_code_path, r"\.idea", "tests", "__pycache__"
+        )
+        recursively_delete_files_by_name(
+            source_code_path,
+            r".*\.pyc",
+            r".*\.pyo",
```

Comment:
we don't have any .pyi files at that moment (type hint stubs), but i guess it probably also doesn't hurt to add it to the ignore here